src/stanc/stanc.ml

```diff
@@ -156,38 +156,11 @@ let add_file filename =
 
 (** ad directives from the given file. *)
 let use_file filename =
-  let ast =
-    try
-      match Parse.parse_file Parser.Incremental.program filename with
-      | Result.Ok ast -> ast
-      | Result.Error err ->
-          let loc = Parse.syntax_error_location err
-          and msg = Parse.syntax_error_message err in
-          Errors.report_parsing_error (msg, loc) ;
-          exit 1
-    with Errors.SyntaxError err ->
-      Errors.report_syntax_error err ;
-      exit 1
-  in
+  let ast = Frontend_utils.get_ast_or_exit filename in
```

Comment:
Not to labor the point but I think we _should_ be doing IO here rather than in a util function in the frontend not least because it means for the 'composable compiler' thing to work it would have to be exposed by all frontends.

Reply:
I'm not sure I get what you mean... the I/O IS being done here. The place that the function it calls lives is irrelevant to the modularity you're espousing (and that I agree with). Happy to put the helpers wherever, though I'm even less sure I understand what you mean now.